File: programmers/[1차] 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    map1 = []
    map2 = []
    
    map1 = convert(n, arr1, map1)
    map2 = convert(n, arr2, map2)
    
    for i in range(n):
        result = ''
        for j in range(n):
            if map1[i][j] == '1' or map2[i][j] == '1':
                result += '#'
            else:
                result += ' '
        answer += [result]
    return answer

# 이진수로 변환하는 함수
def convert(n, arr, map):
    temp = []
    for i in arr:
        result = ''
        while i > 0:
            k = i % 2 # 나머지
            i = i // 2
            result = str(k) + result
        if len(result) != n: # n자리수 맞춰주기
            result = ('0' * (n - len(result))) + result
        temp.append(result)
    return temp

File: programmers/level1/[1차]비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm> // std::reverse

using namespace std;

string decipher(int n, int binary) {
    string result{ "" };
    while (binary > 0) { // 
        result.append(to_string(binary % 2));
        binary /= 2; 
    }
    while (result.length() < n) {
        result.push_back('0'); // 길이 n이 될 때까지 앞에 0 붙임
    }
    reverse(result.begin(), result.end()); // 나머지가 뒤에서부터 저장됨
    return result;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer; // #와 공백
    vector<string> map1, map2;

    for (int i=0; i<n; ++i) {
        map1.push_back(decipher(n, arr1[i]));
        map2.push_back(decipher(n, arr2[i]));
    }

    for (int i=0; i<n; ++i) {
        string row{ "" };
        for (int j=0; j<n; ++j) {
            if (map1[i][j] == '0' && map2[i][j] == '0') {
                row.push_back(' ');
            } else {
                row.push_back('#');
            }
        }
        answer.push_back(row);
    }

    return answer;
}